contracts/AllowedStrategy.sol

```diff
@@ -0,0 +1,270 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity 0.8.11;
+
+import "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+
+import "./common/UsingRegistryUpgradeable.sol";
+import "./common/UUPSOwnableUpgradeable.sol";
+import "./interfaces/IAccount.sol";
+import "./interfaces/IGroupHealth.sol";
+import "./interfaces/IManager.sol";
+import "./Managed.sol";
+
+contract AllowedStrategy is UUPSOwnableUpgradeable, UsingRegistryUpgradeable, Managed {
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    /**
+     * @notice Allowed strategies (validator groups) that can be chosen to voted on.
+     */
+    EnumerableSet.AddressSet private allowedStrategies;
+
+    /**
+     * @notice StCelo that was cast for allowed strategies
+     * strategy => stCelo amount
+     */
+    mapping(address => uint256) private allowedStrategyTotalStCeloVotes;
+
+    /**
+     * @notice Total StCelo that was voted with on allowed strategy
+     */
+    uint256 private totalStCeloInAllowedStrategies;
+
+    /**
+     * @notice Contract used for checking group health and rebalancing.
+     */
+    IGroupHealth public groupHealthContract;
```

Comment:
For consistency with other contract variables throughout the protocol, consider `groupHealth` instead?